.github/workflows/build-connector.yml

```diff
@@ -25,57 +25,55 @@ jobs:
   build-extension:
     name: "Build Connector"
     runs-on: ubuntu-latest
+    if: ${{ !(github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository) }}
+    strategy:
+      matrix:
+        flink-version: ['1.15', '1.16', '1.17', '1.18', '1.19', '1.20']
+        include:
+          - flink-version: '1.15'
+            flink-full-version: '1.15.0'
+            flink-python-id: 'flink-python_2.12'
+          - flink-version: '1.16'
+            flink-full-version: '1.16.0'
+            flink-python-id: 'flink-python'
+          - flink-version: '1.17'
+            flink-full-version: '1.17.0'
+            flink-python-id: 'flink-python'
+          - flink-version: '1.18'
+            flink-full-version: '1.18.0'
+            flink-python-id: 'flink-python'
+          - flink-version: '1.19'
+            flink-full-version: '1.19.0'
+            flink-python-id: 'flink-python'
+          - flink-version: '1.20'
+            flink-full-version: '1.20.0'
+            flink-python-id: 'flink-python'
```

Comment:
I think it is better to put them together. More flink versions will be involved later.